crates/telemetry/src/alert_in_dev.rs

```diff
@@ -0,0 +1,47 @@
+//! Provides a way for warnings to be force-raised in the
+//! `spin up` environment even if RUST_LOG is not set to warn.
+//! This is useful for things that are not errors but where we
+//! want application developers to know they have a problem.
+
+use tracing::{Event, Subscriber};
+use tracing_subscriber::Layer;
+
+const ALERT_IN_DEV_TAG: &str = "alert_in_dev_environment";
+
+/// A layer which prints a terminal warning (using [terminal::warn!]) if
+/// a trace event contains the tag "alert_in_dev_environment".
+pub(crate) fn alert_in_dev_layer<S: Subscriber>() -> impl Layer<S> {
+    CommandLineAlertingLayer
+}
+
+pub struct CommandLineAlertingLayer;
+
+impl<S: Subscriber> Layer<S> for CommandLineAlertingLayer {
+    fn on_event(&self, event: &Event<'_>, _ctx: tracing_subscriber::layer::Context<'_, S>) {
+        let meta = event.metadata().fields();
+        if meta.field(ALERT_IN_DEV_TAG).is_some() {
+            if let Some(m) = extract_message(event) {
+                terminal::warn!("{m}");
+            }
+        }
+    }
+}
+
+fn extract_message(event: &Event<'_>) -> Option<String> {
+    let mut extractor = MessageExtractor::default();
+    event.record(&mut extractor);
+    extractor.message
+}
+
+#[derive(Default)]
+struct MessageExtractor {
+    message: Option<String>,
+}
+
+impl tracing::field::Visit for MessageExtractor {
```

Comment:
I am sure this can't be the correct way to get a value out of an event but I couldn't find a direct way.  If someone would care to educate me on the right way to approach this then I will be forever grateful _(terms and conditions apply)_

Reply:
Oh wait. Maybe the visitor is meant to terminal::warn the thing directly.  Yeah that probably makes more sense